framework/src/kernels/CoupledForce.C

```diff
@@ -32,6 +34,9 @@ validParams<CoupledForce>()
 CoupledForce::CoupledForce(const InputParameters & parameters)
   : Kernel(parameters), _v_var(coupled("v")), _v(coupledValue("v")), _coef(getParam<Real>("coef"))
 {
+  if (_var.number() == _v_var)
```

Comment:
Wait, I though you actually wanted this. I would also accept a PR that properly implements the Jacobian when this case is true. Wouldn't that simplify some of the physics you setup?

Reply:
I do not. Do you?

Reply:
I noticed actually few kernels in test folder need to be fixed similarly. But since they are in tests, we do not have to hopefully...

Reply:
One more thing, let us wait the tests complete to see if anything is broken.